src/shape_type.cpp

```diff
@@ -0,0 +1,43 @@
+#include "shpxx/shape_type.hpp"
+
+#include <string>
+
+namespace shpxx {
+
+std::string to_string(shape_type type) noexcept
+{
+    switch (type)
+    {
+    case shape_type::null:
+        return "null";
```

Comment:
Should I make all these match the enum exactly? e.g. `shpxx::shape_type::null`, `shpxx::shape_type::point`